File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/3.2 Coupling Exercise/After/UserDataService.java
package com.grokkingsoftwarearchitecture.chapter03.couplingexercise.after;

public class UserDataService {
    public UserReportData getUserReport(int userId) {
        System.out.println("    [Service] Building chunky report payload internally...");
        return new UserReportData("Jane Doe", "jane.doe@example.com", 199.90);
    }
}

File: Chapter 03/JS/3.2 Coupling Exercise/After/userDataService.js
class UserDataService {
    getUserReport(userId) {
        console.log("    [Service] Building chunky report payload internally...");
        return {
            name: "Jane Doe",
            email: "jane.doe@example.com",
            totalSpent: 199.90
        };
    }
}

module.exports = UserDataService;